### ISSUE TITLE
Move towards SQLite3 DB backend for local / travis test execution

### ISSUE THREAD

**Issue by dtasev** [ISISScientificComputing/autoreduce#779]
Issue raised by: [developer]

### What?
Currently the tests require a connection to the dev environment database. This is slow, and makes test execution fully locally a lot harder (the developer needs to install their own MySQL)

To do that change the Django DB backend to:
```
DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.sqlite3',
         'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
     }
}
```
Eventually it can even be changed to be stored in a memory file by setting: `'NAME': ':memory:'`

### To do
- [ ] Create fixtures that pre-load the database with data that tests need
  - Things like loading the available instruments in the Instruments table, i.e. GEM, MARI, etc
  - The easiest way to do this may be to get `pytest` to create a `db.sqlite3` file, then go into it and either run the SQL statements from `build/database/populate_reduction_viewer.sql`
- [ ] Fix tests that hard-code `MySQLdb` access (some/all test failures in [this build](https://travis-ci.org/github/ISISScientificComputing/autoreduce/builds/729551201) are caused by the change of database backend - if the build is unavailable, change the backend and run the tests!)
  - I am not sure if this can be moved in a new issue and done in parallel, it does seem likely

### How to test the issue is resolved
`pytest` on root of repository should have 100% pass rate with the SQLite3 back-end, both locally and on CI

**Comment by dtasev** [comment]
Done in #1066
